runtime/interpreter/value.go

```diff
@@ -1133,13 +1089,9 @@ func (v IntValue) ToBigEndianBytes() []byte {
 	return SignedBigIntToBigEndianBytes(v.BigInt)
 }
 
-func (v IntValue) ConformsToDynamicType(dynamicType DynamicType) bool {
+func (v IntValue) ConformsToDynamicType(_ *Interpreter, dynamicType DynamicType) bool {
 	numberType, ok := dynamicType.(NumberDynamicType)
-	return ok && ValueConformsToSemaType(v, numberType.StaticType)
-}
-
-func (v IntValue) conformsToSemaType(semaType sema.Type) bool {
-	return (sema.IntType).Equal(semaType)
+	return ok && numberType.StaticType == sema.IntType
```

Comment:
Maybe we could delegate the static type check to `equal` method
```suggestion
	return ok && sema.IntType.equal(numberType.StaticType)
```

I can add this change.